osu.Game.Rulesets.Mania/Objects/HoldNote.cs

```diff
@@ -1,43 +1,87 @@
 ﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
 // Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
 
-using osu.Game.Audio;
+using System.Collections.Generic;
 using osu.Game.Beatmaps.ControlPoints;
 using osu.Game.Database;
-using osu.Game.Rulesets.Mania.Judgements;
 using osu.Game.Rulesets.Objects.Types;
 
 namespace osu.Game.Rulesets.Mania.Objects
 {
     /// <summary>
     /// Represents a hit object which requires pressing, holding, and releasing a key.
     /// </summary>
-    public class HoldNote : Note, IHasEndTime
+    public class HoldNote : ManiaHitObject, IHasEndTime
     {
-        /// <summary>
-        /// Lenience of release hit windows. This is to make cases where the hold note release
-        /// is timed alongside presses of other hit objects less awkward.
-        /// </summary>
-        private const double release_window_lenience = 1.5;
-
         public double Duration { get; set; }
         public double EndTime => StartTime + Duration;
 
+        private Note head;
+        /// <summary>
+        /// The head note of the hold.
+        /// </summary>
+        public Note Head => head ?? (head = new Note { StartTime = StartTime });
+
+        private Note tail;
         /// <summary>
-        /// The samples to be played when this hold note is released.
+        /// The tail note of the hold.
         /// </summary>
-        public SampleInfoList EndSamples = new SampleInfoList();
+        public Note Tail => tail ?? (tail = new TailNote { StartTime = EndTime });
 
         /// <summary>
-        /// The key-release hit windows for this hold note.
+        /// The time between ticks of this hold.
         /// </summary>
-        public HitWindows ReleaseHitWindows { get; protected set; } = new HitWindows();
+        private double tickSpacing = 50;
 
         public override void ApplyDefaults(ControlPointInfo controlPointInfo, BeatmapDifficulty difficulty)
         {
             base.ApplyDefaults(controlPointInfo, difficulty);
 
-            ReleaseHitWindows = HitWindows * release_window_lenience;
+            TimingControlPoint timingPoint = controlPointInfo.TimingPointAt(StartTime);
+            tickSpacing = timingPoint.BeatLength / difficulty.SliderTickRate;
+        }
+
+        /// <summary>
+        /// The scoring scoring ticks of the hold note.
+        /// </summary>
+        public IEnumerable<HoldNoteTick> Ticks => ticks ?? (ticks = createTicks());
+        private List<HoldNoteTick> ticks;
+
+        private List<HoldNoteTick> createTicks()
+        {
+            var ret = new List<HoldNoteTick>();
+
+            if (tickSpacing == 0)
+                return ret;
+
+            for (double t = StartTime + Head.HitWindows.Great / 2; t <= EndTime - Tail.HitWindows.Great / 2; t+= tickSpacing)

```

Comment:
got any ideas for how to document this? It's a hit_window_, I though that would be explanatory enough? Unlike taiko and osu! I made "hit windows" span the full length (from miss to miss), as we previously discusded. If that's not explanatory enough we need to rename this.